### PR TITLE
[BD-46] feat: implement Chip component

### PR DESCRIPTION
**Pull request by monteri**
added Chip component with tests, description and styles

JIRA: [PAR-437](https://openedx.atlassian.net/jira/software/projects/PAR/boards/654?selectedIssue=PAR-437)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @monteri! I've created [OSPR-6247](https://openedx.atlassian.net/browse/OSPR-6247) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: src/Chip/index.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      {children}
+      {iconAfter && <Icon className={classNames('chip-icon-after')} src={iconAfter} />}
+    </>
+  );
+  return React.createElement(props.href ? 'a' : elementType, elementProps, elementContent);
+};
+
+Chip.propTypes = {
+  /** Specifies the base HTML element. */
+  as: PropTypes.elementType,
```
Comment (viktorrusakov) [inline src/Chip/index.jsx:48]: In the component implementation you only allow 'button' or 'a' elements, maybe you can define this props as `PropTypes.oneOf(['button', 'a'])` or something like this? This way you don't need to check if the prop is correct like you do [here](https://github.com/edx/paragon/pull/911/files#diff-b7460be0253af56f24ad7fb65320ffdac22f6f0f26b87ff93ca36943c3cac36dR26)

I'm also not sure if we need to restrict the element to only 'a' or 'button', I think everywhere else in Paragon `as` is always an arbitrary element, maybe we should do the same here?

File: src/Chip/index.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  disabled: PropTypes.bool,
+  /** Providing a `href` will render an `<a>` element, styled as a button. */
+  href: PropTypes.string,
+  /**
+   * An icon component to render before the content.
+   * Example import of a Paragon icon component:
+   *
+   * `import { Check } from '@edx/paragon/icons';`
+   */
+  iconBefore: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
```
Comment (viktorrusakov) [inline src/Chip/index.jsx:67]: Why do you allow icons to be functions also? 🤔 
Shouldn't it just be `PropTypes.node`?

**Review by viktorrusakov — CHANGES_REQUESTED** [review]

File: src/Chip/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- Chip
+categories:
+- Buttonlike
+status: 'New'
+designStatus: 'Done'
+devStatus: 'In progress'
+notes: |
+---
+
+``Chips`` are compact elements that represent an input, attribute, or action. Similar to the Badge component, but interactive.
```
[the remark this answers is not in the document]
Comment (monteri) [reply src/Chip/README.md:14]: Yes, I will make it is as link to the Badge

File: src/Chip/_variables.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                "dark-background-inactive": $primary-300,
+                "dark-text-inactive":       $white,
+                "dark-border-inactive":     $primary-300,
+                "dark-background-hover":    $white,
+                "dark-text-hover":          $primary-700,
+                "dark-border-hover":        $primary-300,
+                "dark-background-selected": $primary-300,
+                "dark-text-selected":       $white,
+                "dark-border-selected":     $white,
+        );
```
[the remark this answers is not in the document]
Comment (monteri) [reply src/Chip/_variables.scss:36]: Yes, I missed it, thanks

File: src/Chip/index.jsx (hunk 1 above)
Comment (monteri) [reply src/Chip/index.jsx:48]: According to the task it must either `button` or `a`. But I agree, everywhere it is arbitrary and here would be good to have the same behavior. @adamstankiewicz what do you think about this?

Comment (viktorrusakov) [reply src/Chip/index.jsx:48]: Ohh I see, I missed that in task description, sorry.

Let's see what Adam thinks then 🙂

File: src/Chip/index.jsx (hunk 2 above)
Comment (monteri) [reply src/Chip/index.jsx:67]: I copied from the Button component, but yes it should be just `PropTypes.node`. Editing, thanks

File: src/Chip/index.jsx (hunk 1 above)
Comment (adamstankiewicz) [reply src/Chip/index.jsx:48]: Yes, I agree. If we can build this to keep `as` consistent with how it's used every else, we should do that :)

**Review by adamstankiewicz — APPROVED** [review]
This looks good to me. Just a bit of cleanup to make `as` arbitrary like everywhere else from the previous suggestion by @viktorrusakov!

File: src/Chip/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+---
+title: 'Chip'
+type: 'component'
+components:
+- Chip
+categories:
+- Buttonlike
```
[the remark this answers is not in the document]
Comment (adamstankiewicz) [reply src/Chip/README.md:7]: On second thought, I agree. It's interactive as if its a button. Let's keep this as is!

**Comment by edx-webhook** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/paragon&entry.867055334=https://github.com/edx/paragon/pull/911&entry.1484655318=https://github.com/monteri&entry.752974735=2021-11-30+13:58&entry.1917517419=2021-12-16+13:32&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**